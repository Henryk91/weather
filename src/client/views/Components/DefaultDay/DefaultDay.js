import React, { Component } from 'react'
import { Link } from "react-router-dom";
import ReactAnimatedWeather from 'react-animated-weather';
import { convertIconName } from '../../Helpers/helper-functions';

export default class DefaultDay extends Component {
    constructor(props) {
        super(props);
    }
    render() {
        const dailyWeather = this.props.dailyWeather
        const currentWeather = this.props.currentWeather
        let date = new Date(dailyWeather.sunriseTime * 1000).toDateString()
        date = date.substring(0, (date.length - 4))
        return (
            <Link style={{ textDecoration: 'none' }} to={`/detailed/${dailyWeather.time}`}>
                <section key={dailyWeather.time} className="dayBox blueHover">
                    {this.props.id === 0 ?
                        <p className="leadP">Today</p> :
                        <p className="leadP">{date}</p>
                    }
                    <ReactAnimatedWeather
                        icon={convertIconName(dailyWeather.icon)}
                        color={'#FEBF34'}
                        size={40}
                        animate={true}
                    />
                    {this.props.id === 0 ?
                        <p>Current Temp: {currentWeather.temperature}&#176;</p> :
                        null
                    }
                    <p>
                        {Math.round(dailyWeather.temperatureMin)}&#176;/
                        {Math.round(dailyWeather.temperatureMax)}&#176;
                    </p>
                    <p>Rain: {Math.round(dailyWeather.precipProbability * 100)}%</p>
                </section>
            </Link >
        )
    }
}