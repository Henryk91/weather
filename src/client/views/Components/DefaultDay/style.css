.dayBox{
  height: fit-content;
  background: #007998;
  margin: 5px;
  padding: 5px;
  border-radius: 2px;
}
.dayBox :first-child{
  grid-column:1/3;
}
.dayBox > p{
  margin: 0;
  font-size: 14px;
}