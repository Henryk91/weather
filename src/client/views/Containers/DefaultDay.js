import React, { Component } from 'react'
import { Link } from "react-router-dom";
import ReactAnimatedWeather from 'react-animated-weather';



export default class DefaultDay extends Component {
    constructor(props) {
        super(props);
        this.state = {
            data: this.props.data,
            currently: this.props.currently
        }
        this.convertIcon = this.convertIcon.bind(this)
    }

    convertIcon(icon) {
        return icon.toUpperCase().replace(/-/g, "_")
    }
    render() {
        const data = this.state.data
        const currently = this.state.currently
        let date = new Date(data.sunriseTime * 1000).toDateString()
        date = date.substring(0, (date.length -4))
        return (
            <div>
                <Link style={{ textDecoration: 'none' }} to={`/detailed/${data.time}`}>
                
                    <div key={data.time} className="dayBox blueHover" >
                    {this.props.id === 0 ? <p className="leadP">Today</p> : <p className="leadP">{date}</p>}           
                        <ReactAnimatedWeather
                            icon={this.convertIcon(data.icon)}
                            color={'#FEBF34'}
                            size={40}
                            animate={true}
                        />
                        {this.props.id === 0 ? <p >Current Temp: {currently.temperature}&#176;</p> : null}
                        <p>{Math.round(data.temperatureMin)}&#176;/{Math.round(data.temperatureMax)}&#176;</p>
                        <p>Rain: {Math.round(data.precipProbability*100)}%</p>
                    </div>
                </Link>
            </div>
        )
    }
}