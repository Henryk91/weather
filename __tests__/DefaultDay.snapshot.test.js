
import DefaultDay from '../src/client/views/Components/DefaultDay/DefaultDay';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { weatherData } from "../testData/weatherData.sample"

test('DefaultDay component renders all details', () => {
    const wData = weatherData();
    const rendered = renderer.create(
        <Router>
            <DefaultDay weatherData={wData} dailyWeather={wData.daily.data[0]} id={1} />
        </Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
});