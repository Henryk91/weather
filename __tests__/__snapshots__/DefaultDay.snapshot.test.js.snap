// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultDay component renders all details 1`] = `
<div>
  <a
    href="/detailed/1551996000"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <div
      className="dayBox blueHover"
    >
      <p
        className="leadP"
      >
        Fri Mar 08 
      </p>
      <canvas
        height={40}
        width={40}
      />
      <p>
        19
        °/
        30
        °
      </p>
      <p>
        Rain: 
        18
        %
      </p>
    </div>
  </a>
</div>
`;
